src/charm.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+from charms.observability_libs.v0.kubernetes_service_patch import KubernetesServicePatch
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.pebble import Layer
+
+logger = logging.getLogger(__name__)
+
+
+class FluentdElasticsearchCharm(CharmBase):
+
+    CONFIG_DIRECTORY = "/etc/fluent/config.d"
+
+    def __init__(self, *args):
+        """
+        An instance of this object everytime an event occurs
+        """
+        super().__init__(*args)
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.fluentd_elasticsearch_pebble_ready, self._configure)
+        self.framework.observe(self.on.config_changed, self._configure)
+        self._container_name = self._service_name = "fluentd-elasticsearch"
+        self._container = self.unit.get_container(self._container_name)
+        self._service_patcher = KubernetesServicePatch(charm=self, ports=[("fluentd", 24224)])
+
+    def _on_install(self, event):
+        self._write_config_files()
+
+    def _write_config_files(self):
+        base_source_directory = "src/config_files"
+
+        self._write_to_file(
+            source_directory=f"{base_source_directory}/forward-input.conf",
+            destination_directory=f"{self.CONFIG_DIRECTORY}/forward-input.conf",
+        )
+        self._write_to_file(
+            source_directory=f"{base_source_directory}/general.conf",
+            destination_directory=f"{self.CONFIG_DIRECTORY}/general.conf",
+        )
+        self._write_to_file(
+            source_directory=f"{base_source_directory}/output.conf",
+            destination_directory=f"{self.CONFIG_DIRECTORY}/output.conf",
+        )
+        self._write_to_file(
+            source_directory=f"{base_source_directory}/system.conf",
+            destination_directory=f"{self.CONFIG_DIRECTORY}/system.conf",
+        )
+
+    def _write_to_file(self, source_directory: str, destination_directory: str):
+        file = open(source_directory, "r")
+        file_content = file.read()
+        self._container.push(destination_directory, file_content)
```

Comment:
I think that we should close the file, don't we?
```suggestion
        self._container.push(destination_directory, file_content)
        file.close()
```
Or we may want to use `with`

Reply:
Agreed, fixed